wfe2/wfe.go

```diff
@@ -2253,6 +2275,27 @@ func (wfe *WebFrontEndImpl) validateReplacementOrder(ctx context.Context, acct *
 	return replaces, renewalInfo.SuggestedWindow.IsWithin(wfe.clk.Now()), nil
 }
 
+func (wfe *WebFrontEndImpl) validateCertificateProfileName(profile string) error {
+	if profile == "" {
+		// No profile name is specified.
+		return nil
+	}
+	if utf8.RuneCountInString(profile) > 32 {
+		// Profile name will not fit in a VARCHAR(32) column.
+		return errors.New("profile name is too long")
+	}
+	if len(wfe.certificateProfileNames) == 0 {
+		// No certificate profiles are configured, so any profile name is valid.
+		return nil
+	}
```

Comment:
IMO, we remove these two checks. The first one can be handled by the combination of the final check in this function + config validation; the second one because I think we should fail-closed rather than fail-open (see comment above).